proxy/proxy.go

```diff
@@ -837,7 +876,7 @@ func (p *Proxy) makeUpgradeRequest(ctx *context, req *http.Request) {
 }
 
 func (p *Proxy) makeBackendRequest(ctx *context, requestContext stdlibcontext.Context) (*http.Response, *proxyError) {
-	req, endpoint, err := mapRequest(ctx, requestContext, p.flags.HopHeadersRemoval(), p.registry)
+	req, endpoint, err := p.mapRequest(ctx, requestContext, p.flags.HopHeadersRemoval(), p.registry)
```

Comment:
Since you make it a method - drop p.flags.HopHeadersRemoval() and p.registry args